src/cmd_all/src/all_in_one.rs

```diff
@@ -0,0 +1,116 @@
+    let frontend_opts = opts.frontend_opts.split_whitespace().collect::<Vec<_>>();
+    let services = vec![
+        RisingWaveService::Meta(osstrs(&meta_opts)),
+        RisingWaveService::Compute(osstrs(&compute_opts)),
+        RisingWaveService::Frontend(osstrs(&frontend_opts)),
+    ];
+    services
+}
+
+/// TODO: Refactor into utils.
+fn osstrs(s: &[&str]) -> Vec<OsString> {
+    s.iter().map(OsString::from).collect()
+}
+
+pub async fn all_in_one(opts: AllInOneOpts) -> Result<()> {
+    tracing::info!("launching Risingwave in all-in-one mode");
+
+    let services = get_services(&opts);
+
+    for service in services {
+        match service {
+            RisingWaveService::Meta(mut opts) => {
+                opts.insert(0, "meta-node".into());
+                tracing::info!("starting meta-node thread with cli args: {:?}", opts);
+                let opts = risingwave_meta::MetaNodeOpts::parse_from(opts);
+                let _meta_handle = tokio::spawn(async move {
+                    risingwave_meta::start(opts).await;
+                    tracing::warn!("meta is stopped, shutdown all nodes");
+                });
+                // wait for the service to be ready
+                tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+            }
+            RisingWaveService::Compute(mut opts) => {
+                opts.insert(0, "compute-node".into());
+                tracing::info!("starting compute-node thread with cli args: {:?}", opts);
+                let opts = risingwave_compute::ComputeNodeOpts::parse_from(opts);
+                let _compute_handle = tokio::spawn(async move {
+                    risingwave_compute::start(opts, prometheus::Registry::new()).await
+                });
+            }
+            RisingWaveService::Frontend(mut opts) => {
+                opts.insert(0, "frontend-node".into());
+                tracing::info!("starting frontend-node thread with cli args: {:?}", opts);
+                let opts = risingwave_frontend::FrontendOpts::parse_from(opts);
+                let _frontend_handle =
+                    tokio::spawn(async move { risingwave_frontend::start(opts).await });
+            }
+            RisingWaveService::Compactor(_) => {
+                panic!("Compactor node unsupported in Risingwave all-in-one mode.");
+            }
+            RisingWaveService::ConnectorNode(_) => {
+                panic!("Connector node unsupported in Risingwave all-in-one mode.");
+            }
+        }
+    }
+
+    // wait for log messages to be flushed
+    tokio::time::sleep(std::time::Duration::from_millis(100)).await;
+    eprintln!("-------------------------------");
+    eprintln!("RisingWave all-in-one mode is ready.");
+
+    // TODO: should we join all handles?
+    // Currently, not all services can be shutdown gracefully, just quit on Ctrl-C now.
+    // TODO(kwannoel): Why can't be shutdown gracefully? Is it that the service just does not
```

Comment:
Hmm not sure if anything further needs to be done here. I don't have sufficient context about the original comment above this one.

cc @xxchan @BugenZhao

Reply:
Oh seems like `risingwave_rpc_client` does not have graceful exit..

Reply:
Fixing it in separate PR.